### ISSUE TITLE
Panel app does not display in py-editor

### ISSUE THREAD

**Issue by MarcSkovMadsen** [pyscript/pyscript#2118]
### Checklist

- [X] I added a descriptive title
- [X] I searched for other issues and couldn't find a solution or duplication
- [X] I already searched in Google and didn't find any good information or help

### What happened?

Nothing

### What browsers are you seeing the problem on? (if applicable)

Microsoft Edge

### Console info

```shell
No console errors
```

### Additional Context

I hope that one day it would be possible and easy to use pyscript editor with Panel for embedding on web pages etc.

Currently I cannot get it working.

## Reproducible example

**mini-coi.js**

```javascript
/*! coi-serviceworker v0.1.7 - Guido Zuidhof and contributors, licensed under MIT */
/*! mini-coi - Andrea Giammarchi and contributors, licensed under MIT */
(({ document: d, navigator: { serviceWorker: s } }) => {
    if (d) {
      const { currentScript: c } = d;
      s.register(c.src, { scope: c.getAttribute('scope') || '.' }).then(r => {
        r.addEventListener('updatefound', () => location.reload());
        if (r.active && !s.controller) location.reload();
      });
    }
    else {
      addEventListener('install', () => skipWaiting());
      addEventListener('activate', e => e.waitUntil(clients.claim()));
      addEventListener('fetch', e => {
        const { request: r } = e;
        if (r.cache === 'only-if-cached' && r.mode !== 'same-origin') return;
        e.respondWith(fetch(r).then(r => {
          const { body, status, statusText } = r;
          if (!status || status > 399) return r;
          const h = new Headers(r.headers);
          h.set('Cross-Origin-Opener-Policy', 'same-origin');
          h.set('Cross-Origin-Embedder-Policy', 'require-corp');
          h.set('Cross-Origin-Resource-Policy', 'cross-origin');
          return new Response(body, { status, statusText, headers: h });
        }));
      });
    }
  })(self);
```

**config.toml**

```toml
packages = [
    "https://cdn.holoviz.org/panel/1.4.4/dist/wheels/bokeh-3.4.1-py3-none-any.whl",
    "https://cdn.holoviz.org/panel/1.4.4/dist/wheels/panel-1.4.4-py3-none-any.whl"
]
```

**script.html**

```html
<!DOCTYPE html>
<html>
  <head>
    <script src="./mini-coi.js" scope="./"></script>   
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width,initial-scale=1" />
    <title>🦜 Panel Editor Template</title>
    <link rel="stylesheet" href="https://pyscript.net/releases/2024.7.1/core.css">  
    <script type="module" src="https://pyscript.net/releases/2024.7.1/core.js"></script>
  </head>
  <body>
    <script type="py-editor" config="./config.toml">
        import panel as pn

        pn.extension(sizing_mode="stretch_width")
  
        slider = pn.widgets.FloatSlider(start=0, end=10, name='Amplitude')
  
        def callback(new):
           return f'Amplitude is: {new}'
  
        pn.Row(slider, pn.bind(callback, slider)).servable(target="out");
    </script>
    <div id="out"></div>
  </body>
</html>
```

At first I don't see any output in the web page or errors in the console.

![image](https://github.com/user-attachments/assets/19ad14df-9fb7-4e32-84d9-c70366d6ce46)

After click the run button multiple times I see

![image](https://github.com/user-attachments/assets/fda1937d-e470-4ff1-af0a-caedc6613ed1)

I've tried without `target="out"`. I've tried without `.servable(target="out")`. But I cannot get it to display the Panel app.

I would really like the version without `.servable(target="out")` to work as that would be the easiest code to explain to users.

Crossposted on Panel github: https://github.com/holoviz/panel/issues/6991

**Comment by WebReflection** [comment]
have you tried using just `packages = ["bokeh", "panel"]` in the config?

**Comment by MarcSkovMadsen** [comment]
No. Users should be using the links because they are much smaller and known to work with PyScript and Pyodide.

**Comment by WebReflection** [comment]
@MarcSkovMadsen that doesn't answer my question though ... if you use `packages = ["bokeh", "panel"]` does it work?

**Comment by MarcSkovMadsen** [comment]
Ok. first cause identified. Should include panel and bokeh `.js as below

**script.html**

```html
<!DOCTYPE html>
<html>
  <head>
    <script src="./mini-coi.js" scope="./"></script>   
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width,initial-scale=1" />
    <title>🦜 Panel Editor Template</title>
    <script type="text/javascript" src="https://cdn.bokeh.org/bokeh/release/bokeh-3.4.1.js"></script>
    <script type="text/javascript" src="https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.4.1.min.js"></script>
    <script type="text/javascript" src="https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.4.1.min.js"></script>
    <script type="text/javascript" src="https://cdn.jsdelivr.net/npm/@holoviz/panel@1.4.4/dist/panel.min.js"></script
    <link rel="stylesheet" href="https://pyscript.net/releases/2024.7.1/core.css">  
    <script type="module" src="https://pyscript.net/releases/2024.7.1/core.js"></script>
  </head>
  <body>
    <script type="py-editor" config="./config.toml">
        import panel as pn

        pn.extension(sizing_mode="stretch_width")
  
        slider = pn.widgets.FloatSlider(start=0, end=10, name='Amplitude')
  
        def callback(new):
           return f'Amplitude is: {new}'
  
        pn.Row(slider, pn.bind(callback, slider)).servable(target="out");
    </script>
    <div id="out"></div>
  </body>
</html>
```

But now I get 

```bash
Error raised while processing Document events: DataCloneError: Failed to execute 'postMessage' on 'DedicatedWorkerGlobalScope': [object Map] could not be cloned.
```

and `GET https://cdn.holoviz.org/panel/1.4.4/dist/css/loading.css net::ERR_FAILED`

![image](https://github.com/user-attachments/assets/f3337dab-91f9-4077-8303-b0466644f20b)


https://github.com/user-attachments/assets/e6527bbe-2fef-4dc9-90cf-7059b66a2f17

**Comment by MarcSkovMadsen** [comment]
> @MarcSkovMadsen that doesn't answer my question though ... if you use `packages = ["bokeh", "panel"]` does it work?

I just tried. And it gives the same issue.

**Comment by MarcSkovMadsen** [comment]
FYI. I can see that the Panel + PyScript reference example is no longer working. See https://github.com/holoviz/panel/issues/6995.

**Comment by WebReflection** [comment]
py-editor runs in workers ... differently from likely everything you've tried so far out there, as I think only PyScript has the worker ability.

now, because of that, you could:

  * try `<script type="py" config="./config.toml">` instead, no **py-editor**, just **py**
  * if that works, you can try to add a `worker` attribute to that `<script>` ...
  * if that works I can try to find out what's the issue ... if it doesn't, it's possible that panel or bokeh can't work out of a worker so that some *ffi* might be needed to solve, or bigger investigation needs to be done.

I think `from pyscript.ffi import to_js` might solve arguments issues, you should try to wrap that around `pn.Row(to_js(slider), to_js(pn.bind(callback, slider))).servable(target="out")` or one per time see which thing is breaking, 'cause from JS perspective *Map* are in fact cloneable by standard definition and ability of `postMessage` so there's something else wrong in there.

**Comment by MarcSkovMadsen** [comment]
Thx. I'll try. 

Just for info. Panel has been running on pyodide with web workers for a long time (more than a year).

**Comment by WebReflection** [comment]
we have one Pyodide and it worked well in all scenarios ... Pyodide also doesn't use *coincident* behind the scene, AFAIK, so it can't really have sync operations or REPL in there. If there is a way to make panel work in workers we'll figure it out but we should not compare apples to oranges (due different underlying stack)

**Comment by MarcSkovMadsen** [comment]
I tried `<script type="py" config="./config.toml">`. It just does not show anything.

![image](https://github.com/user-attachments/assets/13892ece-48a1-480d-8018-2ee9433e5b81)

```html
<!DOCTYPE html>
<html>
  <head>
    <script src="./mini-coi.js" scope="./"></script>   
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width,initial-scale=1" />
    <title>🦜 Panel Editor Template</title>
    <script type="text/javascript" src="https://cdn.bokeh.org/bokeh/release/bokeh-3.4.1.js"></script>
    <script type="text/javascript" src="https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.4.1.min.js"></script>
    <script type="text/javascript" src="https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.4.1.min.js"></script>
    <script type="text/javascript" src="https://cdn.jsdelivr.net/npm/@holoviz/panel@1.4.4/dist/panel.min.js"></script
    <link rel="stylesheet" href="https://pyscript.net/releases/2024.7.1/core.css">  
    <script type="module" src="https://pyscript.net/releases/2024.7.1/core.js"></script>
  </head>
  <body>
    <script type="py" config="./config.toml">
        import panel as pn

        pn.extension(sizing_mode="stretch_width")
  
        slider = pn.widgets.FloatSlider(start=0, end=10, name='Amplitude')
  
        def callback(new):
           return f'Amplitude is: {new}'
  
        pn.Row(slider, pn.bind(callback, slider)).servable(target="out");
    </script>
    <div id="out"></div>
  </body>
</html>
```

**Comment by MarcSkovMadsen** [comment]
Let me know if there is more I can do. I don't understand the "... wrap that around pn.Row(to_js(slider), to_js(pn.bind(callback, slider))).servable(target="out")" comment.

**Comment by WebReflection** [comment]
you do realize you have a non-closed `</script` tag in your template though ... right?

```html
<script type="text/javascript" src="https://cdn.jsdelivr.net/npm/@holoviz/panel@1.4.4/dist/panel.min.js"></script
```

**Comment by MarcSkovMadsen** [comment]
Thx.

Corrected to

```html
<!DOCTYPE html>
<html>
  <head>
    <script src="./mini-coi.js" scope="./"></script>   
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width,initial-scale=1" />
    <title>🦜 Panel Editor Template</title>
    <script type="text/javascript" src="https://cdn.bokeh.org/bokeh/release/bokeh-3.4.1.js"></script>
    <script type="text/javascript" src="https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.4.1.min.js"></script>
    <script type="text/javascript" src="https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.4.1.min.js"></script>
    <script type="text/javascript" src="https://cdn.jsdelivr.net/npm/@holoviz/panel@1.4.4/dist/panel.min.js"></script>
    <link rel="stylesheet" href="https://pyscript.net/releases/2024.7.1/core.css">  
    <script type="module" src="https://pyscript.net/releases/2024.7.1/core.js"></script>
  </head>
  <body>
    <script type="py" config="./config.toml">
        import panel as pn

        pn.extension(sizing_mode="stretch_width")
  
        slider = pn.widgets.FloatSlider(start=0, end=10, name='Amplitude')
  
        def callback(new):
           return f'Amplitude is: {new}'
  
        pn.Row(slider, pn.bind(callback, slider)).servable(target="out");
    </script>
    <div id="out"></div>
  </body>
</html>
```

But no change. Still does not display anything. Still no errors in console.

**Comment by WebReflection** [comment]
confirmed https://github.com/holoviz/panel/issues/6995#issuecomment-2231486318

will work on this tomorrow ... super weird though, I have no idea why our custom type wouldn't work ... we don't change much in the Python env after all, something is really off here.

**Comment by WebReflection** [comment]
I believe this bug has been fixed both in here (2024.8.2) and upstream. If that's not the case feel free to re-open.
